src/translator.cc

```diff
@@ -129,6 +129,24 @@ namespace ctranslate2 {
     return results;
   }
 
+  void
+  Translator::replace_unknowns(const std::vector<std::string>& source,
+                               std::vector<std::vector<std::string>>& hypotheses,
+                               const std::vector<std::vector<std::vector<float>>>& attention) {
+      for (size_t h = 0; h < hypotheses.size(); ++h) {
+        for (size_t t = 0; t < hypotheses[h].size(); ++t) {
+          if (hypotheses[h][t] == Vocabulary::unk_token) {
+            const std::vector<float> attention_values = attention[h][t];
+            int pos = std::distance(attention_values.begin(),std::max_element(attention_values.begin(),
+                                attention_values.end()));
+
+            std::string source_token = source[pos];
+            hypotheses[h][t] = source_token;
```

Comment:
Avoid the intermediate copy:

```suggestion
            hypotheses[h][t] = source[pos];
```